### PR TITLE
Add cli generated help reference to site

### LINKED ISSUE #4905: Add Docs for Velero commands on the website
Currently, the only way to see information about using the Velero commands is in the command line help. The information there is good, but there is not always room for full examples or details. We should consider adding docs about the commands into the docs site. 

Please upvote this with a thumbs up if you want to see this feature. Downvote it with a thumbs down if you would not use it.

### PR DESCRIPTION
**Pull request by kaovilai**
Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #4905 

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [x] Updated the corresponding documentation in `site/content/docs/main`.

Preview at https://velero-git-site-cli-ref-kaovilai.vercel.app/docs/main/cli-reference
https://velero-git-site-cli-ref-kaovilai.vercel.app/docs/v1.11/cli-reference/
https://velero-git-site-cli-ref-kaovilai.vercel.app/docs/v1.10/cli-reference/
https://velero-git-site-cli-ref-kaovilai.vercel.app/docs/v1.9/cli-reference/

### CODE REVIEW

File: site/layouts/_default/site-header.html [hunk 1]
```diff
@@ -10,6 +10,8 @@
             <li class="blog"><a href="/blog" title="Blog Posts">Blog</a></li>
             <li class="community"><a href="/community" title="Community">Community</a></li>
             <li class="resources"><a href="/resources" title="Resources">Resources</a></li>
+            <li class="cli-reference"><a href="/docs/main/cli-reference" title="CLI Reference">CLI
```
Comment (kaovilai) [inline site/layouts/_default/site-header.html:13]: replace `main` here with {{ .Site.Params.latest }} when next release branch is made with this content.

**Comment by kaovilai** [comment]
This gitignore is causing v1.8 to be ignored
https://github.com/vmware-tanzu/velero/blob/0fa5d2c450e4d97d0623fa9c3da85e7214b6187c/.gitignore#L12

File: site/content/docs/v1.11/cli-reference/velero_backup_create.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  # Wait for a backup to complete before returning from the command.
+  velero backup create backup4 --wait
+```
+
+### Options
+
+```
+      --csi-snapshot-timeout duration                      How long to wait for CSI snapshot creation before timeout.
+      --default-volumes-to-fs-backup optionalBool[=true]   Use pod volume file system backup by default for volumes
```
Comment (kaovilai) [inline site/content/docs/v1.11/cli-reference/velero_backup_create.md:39]: v1.11 option

File: site/content/docs/v1.9/cli-reference/velero_backup_create.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  # Wait for a backup to complete before returning from the command.
+  velero backup create backup4 --wait
+```
+
+### Options
+
+```
+      --csi-snapshot-timeout duration                   How long to wait for CSI snapshot creation before timeout.
+      --default-volumes-to-restic optionalBool[=true]   Use restic by default to backup all pod volumes
```
Comment (kaovilai) [inline site/content/docs/v1.9/cli-reference/velero_backup_create.md:39]: v1.9 option

**Review by kaovilai — COMMENTED** [review]
Notice the generated site is accurate to the release version options

**Comment by reasonerjt** [comment]
@kaovilai thanks for the contribution but I don't think we need to publish this to the website as long as user can see it via `--help`

**Comment by kaovilai** [comment]
From the fix issue it seems some users want to be able to browse around docs and then get linked to CLI references without having to download anything.

It can help drive new users especially those who've yet decided to download the CLI to see what it can do.

**Comment by kaovilai** [comment]
This is generated programmatically and added to the usual `make update` script.

**Pull request closed without merging**